[Sanchita] iP

sannie-beep opened the pull request:
# 🌷͙֒🧚‍♀️TaskFairy - Task Manager Chatbot 🧚‍♀️🌷͙֒🧚‍♀

> “Be curious, don’t retire, and keep very active. Never stop working at something” – Martha Stewart [(source)](https://www.thecorporatesister.com/blog/10-best-quotes-productivity-kickass-women/)

Taskfairy - your all-in-one girly chatbot helper tool to help you achieve your fullest girlboss potential!

She's:
- **text-based**
- *easy to learn*
- ~~complicated~~ SUPERRR FAST to use

### How to Get Started

1. Download her from [here](https://github.com/sannie-beep/ip).
2. Double-click the `.jar` file.
3. Add your tasks.
4. Let TaskFairy magically handle the rest for you! ✨🥰

### Features

- [x] Task Flexibility
   - [x] Todos
   - [x] Deadlines
   - [x] Events
- [x] Management tools
   - [x] Add
   - [x] Delete
   - [x] Mark/Unmark
   - [x] Find

For Java girlies, you can practice your Java skills using it. Here's an example of the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

sannie-beep committed:
rename Duke.java tp Bigmouth.java and implemented skeletal structure

sannie-beep committed:
improved skeletal version of Bigmouth to echo commands and exit when user enters "bye". added new AI.md file to give AI credentials for code generated

sannie-beep committed:
add the ability to store whatever text entered by the user and display them back to the user when requested.

sannie-beep committed:
add the ability to mark tasks as done and add the ability to change the status back to not done.

sannie-beep committed:
added support for tracking 3 types of tasks: todos, deadlines and events

sannie-beep committed:
use the input/output redirection technique to semi-automate the testing of Duke.

sannie-beep committed:
trained the chatbot to deal with errors such as incorrect inputs entered by the user.

sannie-beep committed:
add support for deleting tasks from the list.

sannie-beep committed:
Add save implementation to hard disk txt file

sannie-beep committed:
Add load from hard disk implementation

sannie-beep committed:
Merge branch 'branch-Level-7'

sannie-beep committed:
Teach the chatbot how to understand dates and times.

sannie-beep committed:
Merge branch 'branch-Level-8'

yhanyi commented on src/main/java/Bigmouth.java:
```diff
@@ -0,0 +1,204 @@
+                } else if (userInput.startsWith("mark ")) {
+                    int taskNumber = Integer.parseInt(userInput.split(" ")[1]) - 1;
+                    if (taskNumber < 0 || taskNumber >= tasks.size()) {
+                        throw new BigmouthException("Invalid task number. Please enter a valid task number.");
```
This looks like it might exceed the character limit per line, consider refactoring it!

yhanyi commented on src/main/java/Bigmouth.java:
Consider putting all of your files into a Bigmouth package.

yhanyi commented on text-ui-test/EXPECTED.TXT:
Consider adding more tests to test all your functionalities.

yhanyi commented on src/main/java/Bigmouth.java:
Please add Javadocs to help readers of your code understand it better.

ritvi12 commented on src/main/java/Bigmouth.java:
Can try to put all in one package Bigmouth and then further segregate into different packages like one for tasks and so on.

ritvi12 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,24 @@
+
+    @Override
+    public String toString() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy, h:mma");
```
Can make this a constant variable.

ritvi12 reviewed (commented):
Overall it looks good. There could be some tweaks. Also should add Javadocs wherever necessary to help understand your code better.

sannie-beep committed:
Add More OOP

sannie-beep committed:
Finish adding more OOp

sannie-beep committed:
Organize into packages

sannie-beep committed:
Merge branch 'add-gradle-support'

sannie-beep committed:
Setup Gradle

sannie-beep committed:
Add JUnit Tests

sannie-beep committed:
Add Javadoc header comments to most classes and methods

sannie-beep committed:
Tweak code to comply with Java coding standard

sannie-beep committed:
Implement find task by searching for keyword in task description

sannie-beep committed:
Merge branch 'branch-Level-9'

sannie-beep committed:
Merge branch 'branch-A-CodingStandard' and resolve conflicts

sannie-beep committed:
Merge branch 'branch-A-JavaDoc' and resolve conflicts

sannie-beep committed:
Add GUI to chatbot using JavaFX technology

sannie-beep committed:
Merge branch 'branch-Level-10'

sannie-beep committed:
Use GFMD

sannie-beep committed:
Add assertions to Parser and Storage classes

sannie-beep committed:
Improve code quality and improve readability

* Some classes have methods that are too long and do not use SLAP.
* Additionally, there is too much nesting which makes code less readable
* There is also code that can be simplified using package-protected
constants (Date-time formatting in Deadline and Event Tasks)
* Lastly, there is a lot of dead code to be cleaned up

These classes with complicated methods reduce readability and hence
code quality. Repeated logic due to lack of final static constants and
dead code also reduce readability

Abstracting the logic in long methods to other methods helps make the
code more readable. Furthermore, length of methods can be reduced
through using final constants.

Let's
* abstract the Storage class so that file I/O and task parsing can
be done in separate methods to reduce the size and nesting.
* use a static final constant to represent the formats for Deadline and
Event task parsing and storing

sannie-beep committed:
Add assertions for better error checking

The `Parser` and `Storage` classes handle user inputs and file
operations, respectively. They perform critical tasks like parsing
commands, managing tasks, and saving/loading data, where errors could
lead to incorrect behavior or crashes.

Assertions are missing from these classes, potentially allowing
incorrect assumptions or values to go unnoticed, which can lead to bugs
that are hard to track down.

Let's add assertions to the `Parser` and `Storage` classes to enforce
assumptions and verify critical conditions.
* In the `Storage` class, ensure that paths, task formats, and contents
are valid.
* In the `Parser` class, check that task descriptions are not empty and
that commands are formatted correctly.

Assertions help catch issues early and make debugging easier by
validating expected states and inputs.

sannie-beep committed:
Merge pull request #2 from sannie-beep/branch-A-Assertions

A-Assertions

sannie-beep committed:
Merge branch 'master' of https://github.com/sannie-beep/ip

sannie-beep committed:
Merge branch 'master' into branch-A-CodeQuality

sannie-beep committed:
Merge pull request #3 from sannie-beep/branch-A-CodeQuality

Improve code quality and improve readability

sannie-beep committed:
Merge remote-tracking branch 'origin/master'

sannie-beep committed:
Provide implementation for "Do After" tasks

sannie-beep committed:
Merge branch 'branch-BCD-Extension'

sannie-beep committed:
Update README.md

sannie-beep committed:
Improve the GUI with new color, pictures and font

sannie-beep committed:
Improve personality and use AI to improve testing

sannie-beep committed:
Update AI.md to credit use of generativte AI to assist with project

sannie-beep committed:
Add Ui.png

sannie-beep committed:
Rename Ui.png

sannie-beep committed:
Update README.md

Add a User Guide

sannie-beep committed:
Update README.md

Update on remote repo

sannie-beep committed:
Debug file not found error

sannie-beep committed:
Fix more storage file access, write and load bugs
